pymc/logprob/censoring.py

```diff
@@ -238,3 +260,282 @@ def round_logprob(op, values, base_rv, **kwargs):
+
+    switch_cond, *components = node.inputs
+
+    # deny broadcasting of the switch condition
+    if switch_cond.type.broadcastable != node.outputs[0].type.broadcastable:
+        return None
+
+    measurable_var_switch = [
+        var for var in switch_cond.owner.inputs if check_potential_measurability([var], valued_rvs)
+    ]
+
+    if len(measurable_var_switch) != 1:
+        return None
+
+    current_base_var = measurable_var_switch[0]
+    # deny cases where base_var is some function of 'x', e.g. pt.exp(x), and measurable var in the current switch is x
+    # also check if all the sources of measurability are the same RV. e.g. x1 and x2
+    if current_base_var is not base_rv:
+        return None
+
+    measurable_var_idx = []
+    switch_comp_idx = []
+
+    # get the indices for the switch and other measurable components for further recursion
+    for idx, component in enumerate(components):
+        if check_potential_measurability([component], valued_rvs):
+            if isinstance(
+                component.owner.op, (RandomVariable, SymbolicRandomVariable)
+            ) or not isinstance(component.owner.op.scalar_op, Switch):
+                measurable_var_idx.append(idx)
+            else:
+                switch_comp_idx.append(idx)
+
+    # Check if measurability source and the component itself are the same for all measurable components
+    if any(components[i] is not base_rv for i in measurable_var_idx):
+        return None
+
+    # Get intervals for true and false components from the condition
+    intervals = get_intervals(switch_cond.owner, valued_rvs)
+    adjusted_intervals = adjust_intervals(intervals, outer_interval)
+
+    # Base condition for recursion - when there is no more switch in either of the components
+    if not switch_comp_idx:
+        # Insert the two components and their respective intervals into encoding_list
+        for i in range(2):
+            switch_dict = {
+                "lower": adjusted_intervals[i][0],
+                "upper": adjusted_intervals[i][1],
+                "encoding": components[i],
+            }
+            encoding_list.append(switch_dict)
+
+        return encoding_list
+
+    else:
+        for i in range(2):
+            if i in switch_comp_idx:
+                # Recurse through the switch component(es)
+                encoding_list = flat_switch_helper(
+                    components[i].owner, valued_rvs, encoding_list, adjusted_intervals[i], base_rv
+                )
+
+            else:
+                switch_dict = {
+                    "lower": adjusted_intervals[i][0],
+                    "upper": adjusted_intervals[i][1],
+                    "encoding": components[i],
+                }
+                encoding_list.append(switch_dict)
+
+    return encoding_list
+
+
+@node_rewriter(tracks=[switch])
+def find_measurable_flat_switch_encoding(fgraph: FunctionGraph, node: Node):
```

Comment:
Suggestion, because "flat" is the IR output, not what is being found?

```suggestion
def find_nested_switch_encoding(fgraph: FunctionGraph, node: Node):
```